### PR TITLE
#699 DefaultMavenEnvironmentTest.java:136-143: Ideally integration test is required

### PR DESCRIPTION
**Pull request by dskalenko**
DefaultMavenEnvironmentTest.java:136-143: Ideally integration test is required... #699

### CODE REVIEW

**Comment by davvd** [comment]
@dskalenko Thanks for your pull request, let me find someone who can review it

**Comment by davvd** [comment]
@pinaf this pull request is for you, please review

File: PUZZLES.md [hunk 1]
```diff
@@ -78,5 +78,4 @@ There are 14 active puzzles (77 total):
 &nbsp;&nbsp;&nbsp;&nbsp;[#612](https://github.com/teamed/qulice/issues/612):[**339-f214ff2b**](https://github.com/teamed/qulice/issues/612) [`QulicePlugin.java:47-48`](../master/qulice-gradle-plugin/src/main/java/com/qulice/gradle/QulicePlugin.java#L47-L48)<br/>
 &nbsp;&nbsp;&nbsp;&nbsp;[#411](https://github.com/teamed/qulice/issues/411):[<del>411</del>](https://github.com/teamed/qulice/issues/411) [`?`](../master/?)<br/>
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[#700](https://github.com/teamed/qulice/issues/700):[**411-71c8871e**](https://github.com/teamed/qulice/issues/700) [`CheckstyleValidatorTest.java:383-387`](../master/qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java#L383-L387)<br/>
-&nbsp;&nbsp;&nbsp;&nbsp;[#676](https://github.com/teamed/qulice/issues/676):[<del>676</del>](https://github.com/teamed/qulice/issues/676) [`?`](../master/?)<br/>
```
Comment (pinaf) [inline PUZZLES.md]: @dskalenko why change this file?

**Comment by pinaf** [comment]
@davvd please assign someone else

File: PUZZLES.md (hunk 1 above)
Comment (dskalenko) [inline PUZZLES.md]: @pinaf sorry I didn't know that I can't change this file, if remove/add puzzle - done automatically I'll revert my changes in this file

**Comment by dskalenko** [comment]
@davvd Could you please find someone who can review it

**Comment by davvd** [comment]
> @davvd please assign someone else

@pinaf 15 points was deducted from your rating

**Comment by davvd** [comment]
> @davvd please assign someone else

@pinaf I will ask somebody else to pick this up

**Comment by davvd** [comment]
@dmzaytsev review this one, please

File: PUZZLES.md [hunk 2]
```diff
@@ -79,4 +79,4 @@ There are 14 active puzzles (77 total):
...
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[#700](https://github.com/teamed/qulice/issues/700):[**411-71c8871e**](https://github.com/teamed/qulice/issues/700) [`CheckstyleValidatorTest.java:383-387`](../master/qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java#L383-L387)<br/>
 &nbsp;&nbsp;&nbsp;&nbsp;[#676](https://github.com/teamed/qulice/issues/676):[<del>676</del>](https://github.com/teamed/qulice/issues/676) [`?`](../master/?)<br/>
-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[#699](https://github.com/teamed/qulice/issues/699):[**676-be080182**](https://github.com/teamed/qulice/issues/699) [`DefaultMavenEnvironmentTest.java:136-143`](../master/qulice-maven-plugin/src/test/java/com/qulice/maven/DefaultMavenEnvironmentTest.java#L136-L143)<br/>
\ No newline at end of file
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[#699](https://github.com/teamed/qulice/issues/699):[**676-be080182**](https://github.com/teamed/qulice/issues/699) [`DefaultMavenEnvironmentTest.java:136-143`](../master/qulice-maven-plugin/src/test/java/com/qulice/maven/DefaultMavenEnvironmentTest.java#L136-L143)<br/>
```
Comment (dmzaytsev) [inline PUZZLES.md]: @dskalenko why did you change this file?

**Comment by dmzaytsev** [comment]
@dskalenko  according the puzzle you should revert some changes
> To reproduce problem firstly revert changes marked with #676, 

but I don't see these change or a puzzle for it
correct me if I wrong

File: PUZZLES.md (hunk 2 above)
Comment (dskalenko) [inline PUZZLES.md]: @dmzaytsev I have removed this line by mistake in my previous commit, and now I reverted removal

**Comment by dskalenko** [comment]
@dmzaytsev if I understood correctly it's just steps for reproduce issue #676, but the goal of this issue add integration test which cover issue #676 (again I understood correctly )

File: PUZZLES.md (hunk 2 above)
Comment (dmzaytsev) [inline PUZZLES.md]: @dskalenko the changes doesn't appear in the diff if you reverted the commit

**Comment by dmzaytsev** [comment]
@dskalenko in this case you should create a new puzzle for another developer who will continue you work
see http://www.yegor256.com/2009/03/04/pdd.html

**Comment by dskalenko** [comment]
@dmzaytsev sorry but I didn't get your point, why I have to create a new puzzle? Because goal of issue #699 to complete puzzle, and normally I did it (sorry for misunderstanding and correct me if I am wrong)

**Comment by dmzaytsev** [comment]
@dskalenko is the new integration test fail when you revert changes made by https://github.com/teamed/qulice/pull/696 ?

**Comment by dskalenko** [comment]
@dmzaytsev yes

**Comment by dmzaytsev** [comment]
@dskalenko thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by krzyk** [comment]
@davvd this is not urgent

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by krzyk** [comment]
@rultor deploy

**Comment by davvd** [comment]
> @davvd this is not urgent

@krzyk right, I removed "urgent" tag

**Comment by davvd** [comment]
@dmzaytsev since quality is good, 10 mins paid to @krzyk (our architect), transaction `79409461`

Much obliged! I have added **22 mins** to your account in payment "79409509", 72 hours and 30 mins spent

extra minutes for review comments (c=7)

+22 to your rating, your total score is [+3893](http://www.netbout.com/b/35166?open=rating)

**Comment by davvd** [comment]
@rultor deploy
